PubNub/Network/Parsers/PNSubscribeParser.m

```diff
@@ -152,72 +166,54 @@ + (NSDictionary *)parsedServiceResponse:(id)response withData:(NSDictionary *)ad
+    if ([response isKindOfClass:[NSDictionary class]]) {
+        NSAssert(response[kPNEventsTimeTokenElementKey], @"There should always be a time token object for this key value");
+        NSDictionary *timeTokenDictionary = response[kPNEventsTimeTokenElementKey];
+        NSNumber *timeToken = @([timeTokenDictionary[kPNEventsTimeTokenElementKey] longLongValue]);
+        NSNumber *region = @([timeTokenDictionary[kPNEventsTimeTokenRegionElementKey] integerValue]);
+        NSArray *feedEvents = response[kPNEventsMessageElementKey];
         if ([feedEvents count]) {
-            
             NSMutableArray *events = [[NSMutableArray alloc] initWithCapacity:[feedEvents count]];
             for (NSUInteger eventIdx = 0; eventIdx < [feedEvents count]; eventIdx++) {
                 
                 // Fetching remote data object name on which event fired.
-                NSString *objectOrGroupName = (eventIdx < [channels count] ? channels[eventIdx] : channels[0]);
-                NSString *objectName = ([groups count] > eventIdx ? groups[eventIdx] : nil);
-                NSMutableDictionary *event = [self eventFromData:feedEvents[eventIdx]
-                                                      forChannel:(objectName?: objectOrGroupName)
-                                                           group:(objectName? objectOrGroupName: nil)
-                                        withAdditionalParserData:additionalData];
+                NSMutableDictionary *event = [self eventFromData:feedEvents[eventIdx] withAdditionalParserData:additionalData];
                 event[@"timetoken"] = timeToken;
+                event[@"region"] = region;
                 [events addObject:event];
             }
             feedEvents = [events copy];
         }
-        processedResponse = @{@"events":feedEvents,@"timetoken":timeToken};
+        processedResponse = @{@"events":feedEvents,@"timetoken":timeToken,@"region":region};
     }
-    
+
     return processedResponse;
 }
 
 
 #pragma mark - Events processing
 
-+ (NSMutableDictionary *)eventFromData:(id)data forChannel:(NSString *)channel
-                                 group:(NSString *)group
-              withAdditionalParserData:(NSDictionary *)additionalData {
++ (NSMutableDictionary *)eventFromData:(id)data withAdditionalParserData:(NSDictionary *)additionalData {
     
     NSMutableDictionary *event = [NSMutableDictionary new];
-    if ([channel length]) {
-        
-        event[(![group length] ? @"subscribedChannel": @"actualChannel")] = channel;
+    NSString *channel = data[kPNMessageChannelKey];
+    NSString *subscriptionMatch = data[kPNMessageSubscriptionMatchKey];
+    event[@"subscribedChannel"] = subscriptionMatch;
+    event[@"actualChannel"] = channel;
+    if (data[kPNEventsPublishTimeTokenElementKey] && [data[kPNEventsPublishTimeTokenElementKey] isKindOfClass:[NSDictionary class]]) {
+        event[@"publishTimetoken"] = data[kPNEventsPublishTimeTokenElementKey];
```

Comment:
@jzucker2 event object already have required field for this purpose: _timetoken_ - use it, because it represent time when event occurred (exactly what is stored under _p.t_ key.